docs/reference/imagecraft-yaml-file.rst

```diff
@@ -1,271 +1,182 @@
-.. _imagecraft-yaml-file:
+.. _reference-imagecraft-yaml:
 
-``imagecraft.yaml`` file
-========================
+imagecraft.yaml
+===============
 
-An Imagecraft project is defined in a YAML file named ``imagecraft.yaml``
-at the root of the project tree in the filesystem.
+This reference describes the purpose, usage, and examples of all available keys in
+an image's project file, ``imagecraft.yaml``.
 
-This Reference section is for when you need to know which options can be
-used, and how, in this ``imagecraft.yaml`` file.
 
+Top-level keys
+--------------
 
-name
-----
+An Imagecraft project's top-level keys declare the image's descriptors and the
+essential details of how it builds.
 
-**Type**: string
+Top-level descriptors include the image's name, version, description, and license,
+alongside operational values such as its supported architectures and build environment.
 
-**Required**: Yes
+.. kitbash-field:: craft_application.models.Project name
 
-The name of the image. This value must:
+.. kitbash-field:: craft_application.models.Project title
 
-- start with a lowercase letter [a-z];
-- contain at least one letter;
-- contain only lowercase letters [a-z], numbers [0-9] or hyphens;
-- not end with a hyphen, and must not contain two or more consecutive
-  hyphens.
+.. kitbash-field:: craft_application.models.Project version
 
-title
------
+.. kitbash-field:: craft_application.models.Project license
 
-**Type**: string
+.. kitbash-field:: craft_application.models.Project summary
 
-**Required**: No
+.. kitbash-field:: craft_application.models.Project description
 
-The human-readable title of the image. If omitted, defaults to ``name``.
+.. kitbash-field:: project.Project base
 
-summary
--------
+.. kitbash-field:: project.Project build_base
 
-**Type**: string
+.. kitbash-field:: craft_application.models.Project platforms
```

Comment:
Looks like we need a [docstring for this in craft-application](https://github.com/canonical/craft-application/blob/13abd84e3e3114d8101cfeb95919916776f3bc78/craft_application/models/project.py#L122)